EIPS/eip-4906.md

```diff
@@ -57,7 +58,7 @@ The `supportsInterface` method MUST return `true` when called with `0x49064906`.
 
```

Comment:
Would you mind explaining why not emitting the update event in lines 48-55 is recommended?

Reply:
For line 48: the default behavior of any marketplace I know of would be to read the current metadata from by calling tokenURI(tokenId) once the mint (first transfer event from address(0) to user address is detected.  Emitting MetadataUpdate at that time would be redundant and use a small amount of gas unnecessarily.  The MetadataUpdate is implicitly part of minting.

For line 50: When a token is burned (transfer event to address(0)) the token is considered destroyed - most marketplaces would implicitly hide the destroyed token from view and/or implicitly consider the metadata for the destroyed token to be empty/removed.  Emitting a MetadataUpdate event would be redundant.

For line 52: It is recommended not to notify marketplaces to invoke tokenURI when no functional change has been made to the metadata.  In practice, depending on use case (on-chain vs off-chain metadata, etc) it may be difficult for the contract to determine if a new off-chain URI returns the same or different metadata.  So most likely there will be cases when MetadataUpdate event is emitted for convenience.

Reply:
//  when a token is minted, third parties will get the metadata by tokenURI
// Not emitting `MetadataUpdate` event  helps save gas 
Not emitting `MetadataUpdate` event is RECOMMENDED when a token is minted.

//  when a token is burned, third parties will hidden the NFT or set the owner 
//  to zero address.  And you could not get the  tokenURI for a burned token. 
// Not emitting `MetadataUpdate` event  helps save gas 
Not emitting `MetadataUpdate` event is RECOMMENDED  when a token is burned.

// If the JSON metadata does not change, the NFT properties do not change
// Not emitting `MetadataUpdate` event  helps save gas 
Not emitting `MetadataUpdate` event is RECOMMENDED  when the tokenURI changes but the JSON metadata does not.